### PR TITLE
release: bump version to v0.18.3 rc2

### PR DESCRIPTION
**Pull request by Roasbeef**
Staging PR for the creation of v0.18.3 rc2. 

This PR includes cherry picks of the following PRs:
  * #9002 
  * #9026 
  * #9011 

We also seek to include:
  * https://github.com/lightningnetwork/lnd/pull/9009

PRs on deck for the next rc:
  * https://github.com/lightningnetwork/lnd/pull/9022
     * Fixes an issue in CI, which uncovered additional issues related to the SQL backend. May require another PR to resolve everything we've found so far. 

Once #9009 has been merged, I'll tack on an extra commit that bumps the version.

### CODE REVIEW

**Review by guggero — APPROVED** [review]
LGTM, I checked that the latest PRs merged for the [`v0.18.3` milestone](https://github.com/lightningnetwork/lnd/milestone/50) are included.

File: graph/builder.go [hunk 1]
```diff
@@ -681,7 +681,7 @@ func (b *Builder) handleNetworkUpdate(vb *ValidationBarrier,
 			update.err <- err
 
 		case IsError(err, ErrParentValidationFailed):
-			update.err <- newErrf(ErrIgnored, err.Error())
+			update.err <- NewErrf(ErrIgnored, err.Error()) //nolint
```
Comment (yyforyongyu) [inline graph/builder.go:684]: hmm weird we got this new failure but not in the original pr

Comment (guggero) [reply graph/builder.go:684]: Probably some linter rule changes not in this branch, can likely ignore.

**Comment by Roasbeef** [comment]
So nearly all the SQL backend itests are failing with the latest push. 
* `    --- FAIL: TestLightningNetworkDaemon/tranche06/144-of-168/bitcoind/blinded_payment_htlc_re-forward (69.08s)`
* `        --- FAIL: TestLightningNetworkDaemon/tranche03/80-of-168/bitcoind/multi_hop_htlc_aggregation/zeroconf=false/committype=SCRIPT_ENFORCED_LEASE (171.03s)`
*  ```
    --- FAIL: TestLightningNetworkDaemon/tranche07/156-of-168/bitcoind/listsweeps (74.98s)
    harness_miner.go:195: Mined 7 blocks while waiting for force closed channel to be resolved
    harness.go:442: finished test: stateless_init, start height=586, end height=586, mined blocks=0
    lnd_onchain_test.go:883: 
        	Error Trace:	/home/runner/work/lnd/lnd/itest/lnd_onchain_test.go:883
        	            				/home/runner/work/lnd/lnd/lntest/harness.go:386
        	            				/home/runner/work/lnd/lnd/itest/lnd_test.go:139
        	Error:      	"[9d664884bb1e2edcb61f6d8c55baed65ebb8aa96fd2f47557bc7259faf839d5e a484a7f5366e7e5a933a837b3ddb2b6fa993337ca817655a181c6e7467baa25b]" should have 3 item(s), but has 2
        	Test:       	TestLightningNetworkDaemon/tranche07/156-of-168/bitcoind/listsweeps
        	Messages:   	number of sweeps, starting from height 0
    harness.go:442: finished test: listsweeps, start height=601, end height=644, mined blocks=43
    harness.go:448: test failed, skipped cleanup
    ```
* ```=== RUN   TestLightningNetworkDaemon/tranche06/144-of-168/bitcoind/blinded_payment_htlc_re-forward
    lnd_route_blinding_test.go:1463: 
        	Error Trace:	/home/runner/work/lnd/lnd/itest/lnd_route_blinding_test.go:1463
        	            				/opt/hostedtoolcache/go/1.22.5/x64/src/runtime/asm_amd64.s:1695
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestLightningNetworkDaemon/tranche06/144-of-168/bitcoind/blinded_payment_htlc_re-forward
    harness.go:442: finished test: blinded_payment_htlc_re-forward, start height=1267, end height=1289, mined blocks=22
    harness.go:448: test failed, skipped cleanup
    ```


Re-running failures to gauge if anything introduced in this PR is an issue, or if these are flakes.

**Comment by Roasbeef** [comment]
Looks much better on rerun.

**Pull request merged**